doc/designs/edns.md

```diff
@@ -0,0 +1,198 @@
+# FreeIPA Integrated DNS Service with DoT Support
+
+Encrypted DNS, also known as DNS over HTTPS (DoH) or DNS over TLS (DoT), is a protocol that encrypts DNS queries and responses exchanged between DNS clients (such as web browsers or applications) and DNS servers. Traditional DNS queries are sent over plaintext connections, which can be intercepted and monitored by malicious actors, posing privacy and security risks.
+
+In modern deployments, the internal network can no longer be trusted, it means that all traffic must be authenticated, authorized and encrypted. Encrypted DNS ensures secure communication by enforcing the use of DoT or DoH methods, encrypting all DNS queries and responses. This feature integrates encrypted DNS seamlessly into the FreeIPA management framework, allowing administrators to decide whether the DNS traffic must be encrypted or not.
+
+FreeIPA now includes support for deploying its integrated DNS service with DoT enabled. This enhancement aligns with modern security standards, providing encrypted communication between DNS clients and the FreeIPA DNS server while retaining the same features and benefits of FreeIPA's DNS integration.
+
+## Goals
+
+The primary goal is to provide a way to deploy integrated DNS server with the enforcement of using DNS over encrypted channels instead of using standard UDP port 53 with unencrypted communication. This means that DNS clients must talk to DNS servers using DoT method that is fully encrypted. The scope for the traffic encryption is for all DNS queries that are leaving the host, while the local communication within the host does not require encryption.
+
+
+## Design Choices
+
+This feature builds upon the existing DNS component in FreeIPA, maintaining its core assumptions and goals while introducing secure communication through DoT. It is designed to enhance FreeIPA's DNS deployment options and address the growing need for encrypted DNS communication. This first version of the feature is limited and restricted as we are aiming for a simple implementation that covers basic use cases. These restrictions will be addressed in next iterations of the feature development.
+
+This initial version of DoT is supported only for new installations/deployments, although administrators can run `ipa-dns-install` with the new DoT options to redeploy the integrated DNS service with DoT enabled on an existing deployment.
```

Comment:
The first first part would lead me to believe there was no upgrade path. The second part says you can by re-deploying. I'm not sure the best way to re-word this to be  clearer.

Reply:
fixed.